parser/openfoodfacts_taxonomy_parser/parser/parser.py

```diff
@@ -1,158 +1,213 @@
+        seen_properties = set()
+        seen_tags = set()
+        for entry_node in entry_nodes:
+            if entry_node.get_node_type() != NodeType.ENTRY:
+                raise ValueError(f"Only ENTRY nodes can be batched")
+            for key in entry_node.properties:
+                if not key in seen_properties:
+                    additional_query += " SET n." + key + " = entry_node." + key + "\n"
+                    seen_properties.add(key)
+            for key in entry_node.tags:
+                if not key in seen_tags:
+                    additional_query += " SET n." + key + " = entry_node." + key + "\n"
+                    seen_tags.add(key)
+
+        query = base_query + additional_query
+        self.session.run(query, entry_nodes=[entry_node.to_dict() for entry_node in entry_nodes])
+
+        self.parser_logger.info(
+            f"Created {len(entry_nodes)} ENTRY nodes in {timeit.default_timer() - start_time} seconds"
+        )
+
+    def _create_other_nodes(self, other_nodes: list[NodeData], project_label: str):
+        """Create all TEXT, SYNONYMS and STOPWORDS nodes"""
+        self.parser_logger.info("Creating TEXT, SYNONYMS and STOPWORDS nodes")
+        start_time = timeit.default_timer()
+
+        with self.session.begin_transaction() as tx:
+            for node in other_nodes:
+                self._create_other_node(tx, node, project_label)
 
-    def create_previous_link(self, previous_links: list[PreviousLink], multi_label: str):
+        self.parser_logger.info(
+            f"Created {len(other_nodes)} TEXT, SYNONYMS and STOPWORDS nodes in {timeit.default_timer() - start_time} seconds"
+        )
+
+    def _create_previous_links(self, previous_links: list[PreviousLink], project_label: str):
+        """Create the 'is_before' relations between nodes"""
         self.parser_logger.info("Creating 'is_before' links")
-        for previous_link in previous_links:
-            id = previous_link["id"]
-            before_id = previous_link["before_id"]
-
-            query = f"""
-                MATCH(n:{multi_label}) WHERE n.id = $id
-                MATCH(p:{multi_label}) WHERE p.id= $before_id
-                CREATE (p)-[r:is_before]->(n)
-                RETURN r
-            """
-            results = self.session.run(query, id=id, before_id=before_id)
-            relation = results.values()
-            if len(relation) > 1:
-                self.parser_logger.error(
-                    "2 or more 'is_before' links created for ids %s and %s, "
-                    "one of the ids isn't unique",
-                    id,
-                    before_id,
-                )
-            elif not relation[0]:
-                self.parser_logger.error("link not created between %s and %s", id, before_id)
-
-    def create_child_link(self, child_links: list[ChildLink], multi_label: str):
-        """Create the relations between nodes"""
+        start_time = timeit.default_timer()
+
+        query = f"""
```

Comment:
can you comment a bit on the query.

For example the importance of USING INDEX.